installer/cluster.go

```diff
@@ -370,6 +441,113 @@ func (c *BaseCluster) sshConfig() (*ssh.ClientConfig, error) {
 	return sshConfig, nil
 }
 
+func (c *BaseCluster) attemptSSHConnectionForTarget(t *TargetServer) error {
+	attempts := 0
+	maxAttempts := 30
+	for {
+		var err error
+		t.SSHClient, err = ssh.Dial("tcp", net.JoinHostPort(t.IP, t.Port), t.SSHConfig)
+		if err != nil {
+			if attempts < maxAttempts {
+				attempts++
+				c.SendLog(err.Error())
+				time.Sleep(time.Second)
+				continue
+			}
+			return err
+		}
+		break
+	}
+	return nil
+}
+
+func (c *BaseCluster) uploadBackup() (err error) {
+	if c.backup == nil {
+		return nil
+	}
+
+	// upload to bootstrap instance
+	ipAddress := c.InstanceIPs[0]
+
+	sshConfig, err := c.sshConfig()
+	if err != nil {
+		return err
+	}
+
+	target := &TargetServer{
+		User:      c.SSHUsername,
+		IP:        ipAddress,
+		Port:      "22",
+		SSHConfig: sshConfig,
+	}
+	defer func() {
+		if target.SSHClient != nil {
+			target.SSHClient.Close()
+		}
+
+		c.backupErrChan <- err
+	}()
+
+	return c.uploadBackupToTarget(target)
+}
+
+func (c *BaseCluster) uploadBackupToTarget(t *TargetServer) error {
+	c.SendLog("Uploading backup")
+
+	c.SendProgress(&ProgressEvent{
+		ID:          "upload-backup",
+		Description: "Upload starting...",
+		Percent:     0,
+	})
+
+	if t.SSHClient == nil {
+		if err := c.attemptSSHConnectionForTarget(t); err != nil {
+			return err
+		}
+	}
+
+	sess, err := t.SSHClient.NewSession()
+	if err != nil {
+		return err
+	}
+	stdin, err := sess.StdinPipe()
+	if err != nil {
+		return err
+	}
+	sess.Stdout = os.Stdout
+	sess.Stderr = os.Stderr
+
+	c.backupPath = "/tmp/flynn-backup.tar"
+	if err := sess.Start(fmt.Sprintf("tee > %s", c.backupPath)); err != nil {
```

Comment:
Why `tee` instead of `cat`? Won't that end up downloading it back down from the server?